document data export (#2756)
giantswarm/docs: src/content/overview/observability/data-management/data-export/index.md

---

> title: Data Export

Reviewer: I'm wondering - since it's not just talking about exporting data in here but also ingesting data, maybe the title is not quite right or at least missing something, wdyt?

description: Learn how to export observability data from the Giant Swarm platform using the Observability Platform API for external analysis and integration.
weight: 50
menu:
  principal:
    parent: overview-observability-data-management
    identifier: overview-observability-data-management-data-export
last_review_date: 2025-03-05
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
user_questions:
  - How do I export data from the observability platform?
  - How can I access observability data externally?
  - How do I connect external tools to the platform?
  - What is the Observability Platform API?
  - How do I set up external Grafana to access Giant Swarm data?
---

Data export enables you to access your observability data from external systems and tools, giving you the flexibility to integrate Giant Swarm's observability platform with your existing monitoring infrastructure or specialized analysis tools.

The Observability Platform API serves as the primary mechanism for data export, providing secure, authenticated access to your metrics, logs, and events from anywhere - not just from within Giant Swarm managed clusters.

> ## Why export data?

Reviewer 2: ```suggestion
## Why exporting data
```

- **Multi-cloud strategies**: Centralize observability data from multiple cloud providers and platforms

## How data export works

The Observability Platform API provides both **data ingestion** (sending data to the platform) and **data export** (retrieving data from the platform) capabilities through a unified, secure interface.

### Architecture overview

The API consists of different ingress components that use:

- **Shared host**: Based on your Giant Swarm installation's base domain (`https://observability.<domain_name>`)
- **OIDC authentication**: Secure access through your identity provider
- **Multi-tenant access control**: Tenant-scoped data access through HTTP headers

![Data export architecture](./observability-platform-api-graph.png)
[_Full size architecture diagram_](./observability-platform-api-graph-big.png)

### Authentication and access control

All data export requests require:

1. **Valid OIDC token**: Authentication through your organization's identity provider
2. **Tenant specification**: Include an `X-Scope-OrgId` HTTP header with an existing tenant name

> 3. **Proper authorization**: Your identity must have access to the specified tenant

Reviewer 2: ```suggestion

Please note that your identity must have access to the specified tenant.
```
Since it is not part of the request, it is a precondition

> ⚠️ **Important**: Only data from tenants defined in [Grafana Organization]({{< relref "/overview/observability/configuration/multi-tenancy/" >}}) resources can be accessed. Requests for non-existent tenants will be rejected.

Reviewer: ```suggestion
⚠️ **Important**: Only data from tenants defined in [Grafana Organization]({{< relref "/overview/observability/configuration/multi-tenancy/" >}}) resources can be accessed or ingested. Requests for non-existent tenants will be rejected.
```

## Available data types

> The platform supports exporting different types of observability data:

Reviewer: ```suggestion
The platform supports exporting or ingesting different types of observability data:
```

I'm not sure if we should mix that here or just have another doc. I feel like having an additional doc would be too much overhead. Maybe we just change this doc to cover "Export and Import", while importing is the data ingestion through the API? 
Then we need to make sure the text is adapted everywhere. wdyt?

- **Audit logs**: Security and compliance-related events

### Metrics 🚧

Metrics export capabilities are in development:

- **Infrastructure metrics**: CPU, memory, disk, and network metrics
- **Application metrics**: Custom business and performance metrics
- **Platform metrics**: Kubernetes and Giant Swarm platform metrics

## Export methods

### Method 1: External Grafana integration

Connect your self-managed Grafana instance to access Giant Swarm observability data through familiar dashboards and queries.

#### Setting up Grafana data sources

1. **Configure the connection URL**:
   - For logs (Loki): `https://observability.<domain_name>`
   - For metrics (Mimir/Prometheus): `https://observability.<domain_name>/prometheus`

   Replace `<domain_name>` with your installation's base domain.

   ![Data source URL configuration](./datasource-url.png)

2. **Set up authentication**:
   - Select "Forward OAuth Identity" in the Authentication section
   - This passes your OIDC credentials to the API

   ![Data source authentication setup](./datasource-authentication.png)

3. **Configure tenant access**:
   - Add an `X-Scope-OrgID` custom header

>    - Set the value to your target tenant (e.g., `giantswarm` for platform logs, `anonymous` for platform metrics)

Reviewer 2: ```suggestion
   - Set the value to your target tenant (for example, `giantswarm` for platform logs, `anonymous` for platform metrics)
```

- **Implement caching**: Cache frequently accessed data in your external systems
- **Schedule intensive exports**: Run large data exports during off-peak hours
- **Monitor impact**: Track export performance and adjust patterns as needed

## Troubleshooting

### Common issues

**Authentication failures**:

- Verify OIDC token validity and refresh if expired
- Confirm your identity has access to the specified tenant
- Check that the tenant exists in a Grafana Organization

**Data access issues**:

- Verify the `X-Scope-OrgId` header matches an existing tenant
- Confirm the tenant contains the data you're trying to access
- Check that data ingestion is properly configured for your tenant

**Performance problems**:

- Reduce query time ranges or add more specific filters
- Verify network connectivity and latency to the API endpoints
- Check platform status and resource utilization

### Getting help

For data export setup and troubleshooting:

- **Documentation**: Review [multi-tenancy configuration]({{< relref "/overview/observability/configuration/multi-tenancy/" >}}) for tenant management
- **Support**: Contact Giant Swarm support for authentication and API setup assistance

> - **Community**: Join discussions about observability integrations and best practices

Author: I removed them
